scripts/config_manage.py

```diff
@@ -0,0 +1,473 @@
+    parser.add_argument('--add_comments', default=False, action="store_true")
+    args = parser.parse_args()
+    app_name = args.app
+    app_desc = APPS.get(app_name, None)
+    action = args.action
+    if action == "convert":
+        _run_conversion(args, app_desc)
+    elif action == "build_sample_yaml":
+        _build_sample_yaml(args, app_desc)
+    elif action == "validate":
+        _validate(args, app_desc)
+    elif action == "build_uwsgi_yaml":
+        _build_uwsgi_schema(args, app_desc)
+    elif action == "build_rst":
+        _to_rst(args, app_desc)
+    else:
+        raise Exception("Unknown config action [%s]" % action)
+
+
+def _to_rst(args, app_desc, heading_level="~"):
+    rst = StringIO()
+    schema = app_desc.schema
+    for key, value in schema.app_schema.items():
+        default = None if "default" not in value else value["default"]
+        option = schema.get_app_option(key)
+        option_value = OptionValue(key, default, option)
+        _write_option_rst(args, rst, key, heading_level, option_value)
+    print(rst.getvalue())
+
+
+def _write_option_rst(args, rst, key, heading_level, option_value):
+    title = "``%s``" % key
+    heading = heading_level * len(title)
+    rst.write("%s\n%s\n%s\n\n" % (heading, title, heading))
+    option, value = _parse_option_value(option_value)
+    desc = option["desc"]
+    rst.write(":Description:\n")
+    rst.write("\n".join(RST_DESCRIPTION_WRAPPER.wrap(desc)))
+    rst.write("\n")
+    type = option.get("type", None)
+    default = option.get("default", "*null*")
+    rst.write(":Default: %s\n" % default)
+    if type:
+        rst.write(":Type: %s\n" % type)
+    rst.write("\n\n")
+
+
+def _build_uwsgi_schema(args, app_desc):
+    req = urllib2.Request('https://raw.githubusercontent.com/unbit/uwsgi-docs/master/Options.rst')
+    response = urllib2.urlopen(req)
+    rst_options = response.read()
+    last_line = None
+    current_opt = None
+
+    options = OrderedDict({})
+    option = None
+    for line in rst_options.splitlines():
+        line = line.strip()
+        dots = "*" * len(line)
+        if line and (line == dots):
+            current_opt = last_line
+            option = {
+                'type': 'any',
+            }
+            options[current_opt] = option
+
+        if line.startswith("``parser``"):
+            parser = line.split(":", 1)[1].strip()
+            if parser == "uwsgi_opt_set_int":
+                option["type"] = "int"
+            # TODO: disptch on parser...
+        elif line.startswith("``help``"):
+            option["desc"] = line.split(":", 1)[1]
+
+        last_line = line
+    schema = {
+        "type": "map",
+        "desc": "uwsgi definition, see http://uwsgi-docs.readthedocs.io/en/latest/Options.html",
+        "mapping": options
+    }
+    with open(UWSGI_SCHEMA_PATH, "w") as f:
+        _ordered_dump(schema, f)
+
+
+def _validate(args, app_desc):
+    path = app_desc.destination
```

Comment:
Validate currently only works on the YAML files, I'll see if I can make that more clear.